RUFAS/input_manager.py

```diff
@@ -2221,25 +2225,252 @@ def _validate_properties(self) -> None:
+                raise ValueError(f"Invalid 'default' for '{key_path}': 'default' {default} is "
+                                 f"greater than 'maximum' {maximum}")
 
-    def _metadata_string_validator(self, key_path: List[str], value: dict[str, Any]) -> None:
-        """Validator function for string type properties in metadata."""
-        pass
+    def _metadata_string_validator(self, key_path: list[str], value: dict[str, Any]) -> None:
+        """Validates string type properties in metadata."""
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self._metadata_string_validator.__name__,
+        }
+        required_str_property_keys = {"type"}
+        optional_str_property_keys = {"description", "pattern", "default", "nullable"}
+        self._validate_metadata_properties_keys(required_str_property_keys, optional_str_property_keys, value, key_path)
+        default = value.get("default", "No default")
+        has_no_default = default == "No default"
+        nullable = value.get("nullable", False)
+        if default is None and not nullable:
+            om.add_error(
+                "Invalid metadata default string value.",
+                f"Invalid 'default' for '{key_path}': Value is not nullable and default is 'None'",
+                info_map,
+            )
+            raise ValueError(f"Invalid 'default' for '{key_path}': Value is not nullable and default is 'None'")
+        if default is not None and not has_no_default:
+            if not isinstance(default, str):
+                om.add_error(
+                    "Invalid metadata default string value.",
+                    f"Invalid 'default' for '{key_path}': Expected a string but got {type(default)}",
+                    info_map,
+                )
+                raise ValueError(f"Invalid 'default' for '{key_path}': Expected a string but got {type(default)}")
+        pattern = value.get("pattern")
+        if pattern is not None and not isinstance(pattern, str):
+            om.add_error(
+                "Invalid metadata string properties pattern.",
+                f"Invalid 'pattern' for '{key_path}': Expected a string but got {type(pattern)}",
+                info_map,
+            )
+            raise ValueError(f"Invalid 'pattern' for '{key_path}': Expected a string but got {type(pattern)}")
+        try:
+            if pattern is not None:
+                re.compile(pattern)
+        except re.error:
+            om.add_error(
+                "Invalid metadata string properties pattern.",
+                f"Invalid 'pattern' for '{key_path}': 'pattern' value '{pattern}' is not "
+                "a valid regex pattern.",
+                info_map,
+            )
+            raise ValueError(f"Invalid 'pattern' for '{key_path}': 'pattern' value '{pattern}' is not "
+                             "a valid regex pattern.")
+        if default != "" and default is not None and not has_no_default:
```

Comment:
```suggestion
        if default != "" and default is not None and not has_no_default:
```
If the default is an empty string, shouldn't it still be checked against pattern to make sure that it is a valid default?

Reply:
Unfortunately an empty string doesn't match a regex pattern that specifically requires at least one character to be present. This mostly came up because of the default for `output_pool_path` in `tasks_properties.json` is an empty string. It needs to have a pattern because if a path is given, it needs to be a valid path. So I'm not sure the best way around it.

```
"output_pool_path": {
          "type": "string",
          "pattern": "^(?:[a-zA-Z]:[\\/])?(?:[a-zA-Z0-9._-]+[\\/])*(?:[a-zA-Z0-9._-]+\\.json)$",
          "description": "The path to a JSON files to load the Output Manager's pool from for post processing",
          "default": ""
        },
```

Any suggestions?